src/index.ts

```diff
@@ -57,26 +57,47 @@ export const webpack = (
 
   logger.info(`=> Using PostCSS preset with postcss@${version}`);
 
+  const { rules = [] } = webpackConfig.module || {};
+
+  const prevCssRuleIdx = rules.findIndex(
+    ({ test }) => test?.toString() === '/\\.css$/',
+  );
+  const cssRuleIdx = prevCssRuleIdx === -1 ? rules.length : prevCssRuleIdx;
+  const rulesBefore = rules.slice(0, cssRuleIdx);
+  const rulesAfter = rules.slice(cssRuleIdx + 1);
+
+  const cssRule = {
+    test: /\.css$/,
+    sideEffects: true,
+    exclude: /\.module\.css$/,
+    ...rule,
+    use: [
+      ...wrapLoader(require.resolve('style-loader'), styleLoaderOptions),
+      ...wrapLoader(require.resolve('css-loader'), cssLoaderOptions),
+      ...wrapLoader(require.resolve('postcss-loader'), postcssLoaderOptions),
+    ],
+  };
+
+  const cssModulesRule = {
+    test: /\.css$/,
+    include: /\.module\.css$/,
+    sideEffects: true,
+    ...rule,
+    use: [
+      ...wrapLoader(require.resolve('style-loader'), styleLoaderOptions),
+      ...wrapLoader(require.resolve('css-loader'), {
+        ...cssLoaderOptions,
+        modules: true,
```

Comment:
I would move `modules: true` to above the spread so it's easier to set the module options, like so:       
```js
[
  {
    name: "@storybook/addon-postcss",
    options: {
      postcssLoaderOptions: {
        implementation: require("postcss"),
      },
      cssLoaderOptions: {
        modules: {
          localIdentName: "[path][name]__[local]",
        },
      },
    },
  },
]
```